### PR TITLE
Restart server on each failed test in worker

### LINKED ISSUE #137: Reschedule worker queue if test hangs
It is inconvinient when some tests do not executed by wrotker is it stails, i hope it are able to reschedule to give chance to execute for each test in suite

### LINKED ISSUE #192: Worker fails to work after previously failed test with SIGABRT
Found that after any test previously failed with SIGABRT:
```
2019-11-18 14:01:17.263 [6705] main/113/applier/cluster@unix/:/source/t coio.cc:340 !> SystemError timed out: Connection timed out
[033] replication/autobootstrap.test.lua              memtx           
[033] 
[033] [Instance "autobootstrap1" killed by signal: 6 (SIGABRT)]
```
 the next test on the current worker fails to start tarantool:
```
[033] -test_run = require('test_run').new()
[033] +[Lost current language connection]
```
Error log:
```
[013] sql-tap/whereF.test.lua                         vinyl           [ pass ]
2019-11-18 14:01:17.263 [6705] main/112/applier/cluster@unix/:/source/t I> will retry every 0.10 second
2019-11-18 14:01:17.263 [6705] main/113/applier/cluster@unix/:/source/t I> can't read row
2019-11-18 14:01:17.263 [6705] main/113/applier/cluster@unix/:/source/t coio.cc:340 !> SystemError timed out: Connection timed out
[033] replication/autobootstrap.test.lua              memtx           
[033] 
[033] [Instance "autobootstrap1" killed by signal: 6 (SIGABRT)]
[033] Found assertion fail in the results file [/source/test/var/033_replication/autobootstrap1.log]:
2019-11-18 14:01:17.264 [6705] main/113/applier/cluster@unix/:/source/t I> will retry every 0.10 second
2019-11-18 14:01:18.213 [6705] main/113/applier/cluster@unix/:/source/t I> authenticated
2019-11-18 14:01:18.222 [6705] main/113/applier/cluster@unix/:/source/t I> subscribed
2019-11-18 14:01:18.222 [6705] main/113/applier/cluster@unix/:/source/t I> remote vclock {1: 11, 2: 1, 3: 19, 4: 312} local vclock {1: 11, 2: 1, 3: 19, 4: 311}
2019-11-18 14:01:18.223 [6705] main txn.c:446 W> too long WAL write: 1 rows at LSN 343: 0.550 sec
2019-11-18 14:01:18.224 [6705] main/113/applier/cluster@unix/:/source/t I> can't read row
2019-11-18 14:01:18.225 [6705] main/113/applier/cluster@unix/:/source/t coio.cc:340 !> SystemError timed out: Connection timed out
2019-11-18 14:01:18.226 [6705] main/113/applier/cluster@unix/:/source/t I> will retry every 0.10 second
2019-11-18 14:01:18.227 [6705] main/112/applier/cluster@unix/:/source/t I> authenticated
2019-11-18 14:01:18.243 [6705] main/112/applier/cluster@unix/:/source/t I> subscribed
2019-11-18 14:01:18.243 [6705] main/112/applier/cluster@unix/:/source/t I> remote vclock {1: 11, 2: 1, 3: 19, 4: 313} local vclock {1: 11, 2: 1, 3: 19, 4: 312}
tarantool: /source/src/box/vclock.c:43: int64_t vclock_follow(struct vclock *, uint32_t, int64_t): Assertion `lsn >= 0' failed.
[044] sql-tap/join6.test.lua                          vinyl           [ pass ]
[044] sql-tap/tkt3791.test.lua                        memtx           [ disabled ]
[043] wal_off/lua.test.lua                                            [ pass ]
[040] sql-tap/tkt-7a31705a7e6.test.lua                memtx           [ pass ]
[026] sql-tap/select4.test.lua                        vinyl           [ pass ]
[042] sql-tap/join4.test.lua                          memtx           [ pass ]
[036] replication/autobootstrap_guest.test.lua        memtx           [ pass ]
[038] replication/replicaset_ro_mostly.test.lua       vinyl           [ pass ]
[033] [ fail ]
[033] replication/show_error_on_disconnect.test.lua   memtx           [ fail ]
[033] 
[033] Test failed! Result content mismatch:
[033] --- replication/show_error_on_disconnect.result	Mon Nov 18 13:18:21 2019
[033] +++ replication/show_error_on_disconnect.reject	Mon Nov 18 14:01:23 2019
[033] @@ -1,15 +1 @@
[033] ---
[033] --- gh-3365: display an error in upstream on downstream failure.
[033] --- Create a gap in LSN to cause replica's failure.
[033] --- The goal here is to see same error message on both side.
[033] ---
[033] -test_run = require('test_run').new()
[033] ----
[033] -...
[033] -SERVERS = {'master_quorum1', 'master_quorum2'}
[033] ----
[033] -...
[033] --- Deploy a cluster.
[033] -test_run:create_cluster(SERVERS)
[033] ----
[033] -...
[033] +[Lost current language connection]
[033] 
[033] Last 15 lines of Tarantool Log file [Instance "master"][/source/test/var/033_replication/master.log]:
[033] 2019-11-18 14:01:15.750 [5346] main/159/console/unix/: I> autobootstrap1 -> autobootstrap2: waiting for connection
[033] 2019-11-18 14:01:15.784 [5346] main/159/console/unix/: I> connecting autobootstrap1 -> autobootstrap2: {"status":"follow"}
[033] 2019-11-18 14:01:15.784 [5346] main/159/console/unix/: I> autobootstrap1 -> autobootstrap2: connected
[033] 2019-11-18 14:01:15.784 [5346] main/159/console/unix/: I> autobootstrap3 -> autobootstrap2: waiting for connection
[033] 2019-11-18 14:01:15.813 [5346] main/159/console/unix/: I> connecting autobootstrap3 -> autobootstrap2: {"status":"follow"}
[033] 2019-11-18 14:01:15.813 [5346] main/159/console/unix/: I> autobootstrap3 -> autobootstrap2: connected
[033] 2019-11-18 14:01:15.813 [5346] main/159/console/unix/: I> autobootstrap3: waiting bootstrap
[033] 2019-11-18 14:01:15.825 [5346] main/159/console/unix/: I> autobootstrap3: bootstrapped
[033] 2019-11-18 14:01:15.825 [5346] main/159/console/unix/: I> autobootstrap1 -> autobootstrap3: waiting for connection
[033] 2019-11-18 14:01:15.869 [5346] main/159/console/unix/: I> connecting autobootstrap1 -> autobootstrap3: {"status":"follow"}
[033] 2019-11-18 14:01:15.870 [5346] main/159/console/unix/: I> autobootstrap1 -> autobootstrap3: connected
[033] 2019-11-18 14:01:15.870 [5346] main/159/console/unix/: I> autobootstrap2 -> autobootstrap3: waiting for connection
[033] 2019-11-18 14:01:15.956 [5346] main/159/console/unix/: I> connecting autobootstrap2 -> autobootstrap3: {"status":"follow"}
[033] 2019-11-18 14:01:15.956 [5346] main/159/console/unix/: I> autobootstrap2 -> autobootstrap3: connected
[033] 2019-11-18 14:01:15.956 [5346] main/159/console/unix/: I> full mesh connected
[033] replication/consistent.test.lua                 memtx           [ disabled ]
[044] sql-tap/select7.test.lua                        vinyl           [ pass ]
```

### LINKED ISSUE #215: Investigate whether it is possible to restart an instance within each worker queue
Measure full test suite with different threads count, say: 1, 8, 16, 32, 64.

### PR DESCRIPTION
**Pull request by avtikhon**
Found that non restartable workers, used to run groups of tasks, had
only single tarantool server process started in the very start of it
at the worker's initalization. If any test in the worker's queue was
aborted then all the rest tests in queue couldn't be started. To fix
it the server restart commands were added just after the each task
failed. This change could decrease the performance, but it did not,
because in parallel runs these servers restarts don't use resources
and in absolute testing time was not increased. It happened because
there were other bottle necks in resources that made this change
invisible for the performance. Checking the change w/o parallel runs
found that even in this mode the running time wasn't increased too.

Closes #192
Closes #137
Closes #215

### CODE REVIEW

File: lib/worker.py [hunk 1]
```diff
@@ -353,11 +358,17 @@ def run_loop(self, task_queue, result_queue):
...
             result_queue.put(self.wrap_result(task_id, short_status, result_checksum))
-            if not lib.Options().args.is_force and short_status == 'fail':
-                color_stdout(
-                    'Worker "%s" got failed test; stopping the server...\n'
-                    % self.name, schema='test_var')
-                raise VoluntaryStopException()
+            if short_status == 'fail':
+                if lib.Options().args.is_force:
+                    self.restart_server(task_queue)
+                    color_stdout(
```
Comment (LeonidVas) [inline lib/worker.py:364]: Prints in both cases, just move it up (before if...else).

File: lib/worker.py [hunk 2]
```diff
@@ -285,6 +285,11 @@ def task_get(self, task_queue):
     def is_joinable(task_queue):
         return 'task_done' in task_queue.__dict__.keys()
 
+    def restart_server(self, task_queue):
```
Comment (LeonidVas) [inline lib/worker.py:288]: task_queue passed but not used.

**Review by LeonidVas — CHANGES_REQUESTED** [review]
I am concerned about artifacts, please calm me down. If two tests on the worker fail (one before the server restart and one after), will we have artifacts for both?

**Comment by Totktonada** [comment]
NB: b43ecd1c2e2a4c3d24832ddd7fbbf1a895512cd2.

File: lib/worker.py (hunk 1 above)
Comment (avtikhon) [reply lib/worker.py:364]: "restart" message is the new one and only for "force" tests mode. Message "stop" was never printed before for force mode, also with "restart" message it's not needed.

File: lib/worker.py (hunk 2 above)
Comment (avtikhon) [reply lib/worker.py:288]: Ok.

**Comment by avtikhon** [comment]
> I am concerned about artifacts, please calm me down. If two tests on the worker fail (one before the server restart and one after), will we have artifacts for both?

Right, artifacts will be saved for both. Check that stop call is with cleanup disabled in restart function:
    def restart_server(self):
        self.stop_server(cleanup=False)

**Review by LeonidVas — APPROVED** [review]
LGTM.

**Pull request merged**